gene2phenotype_project/gene2phenotype_app/serializers.py

```diff
@@ -974,30 +1420,42 @@ class PhenotypeSerializer(serializers.ModelSerializer):
     name = serializers.CharField(source="term", read_only=True)
     description = serializers.CharField(read_only=True)
 
-    @transaction.atomic
-    def create(self, validated_data):
-        phenotype_accession = validated_data.get('accession')
+    def create(self, accession):
+        """
+            Create a phenotype based on the accession.
+        """
+        phenotype_accession = accession["accession"]
         phenotype_description = None
 
-        # Check if accession is valid
+        # Check if accession is valid - query HPO API
         validated_phenotype = validate_phenotype(phenotype_accession)
 
         if not re.match(r'HP\:\d+', phenotype_accession) or validated_phenotype is None:
-            raise serializers.ValidationError({"message": f"invalid phenotype accession",
-                                               "please check id": phenotype_accession})
+            raise serializers.ValidationError({"message": f"Invalid phenotype accession",
+                                               "Please check ID": phenotype_accession})
+
+        # TODO check if the new API has 'isObsolete'
+        # if validated_phenotype['isObsolete'] == True:
+        #     raise serializers.ValidationError({"message": f"Phenotype accession is obsolete",
+        #                                        "Please check id": phenotype_accession})
```

Comment:
if this code is not required, please remove it

Reply:
I suspect there are some obsolete phenotypes but as I couldn't find any example I commented the code to be used in the future.